functions/Get-DbaBackupHistory.ps1

```diff
@@ -236,42 +227,41 @@ Lots of detailed information for all databases on sqlserver2014a and sql2016.
-									  mediafamily.physical_device_name AS Path,
-									  CAST((backupset.$BackupSizeColumn / 1048576) AS numeric(10, 2)) AS TotalSizeMB,
-									  CASE backupset.type
-										WHEN 'L' THEN 'Log'
-										WHEN 'D' THEN 'Full'
-										WHEN 'F' THEN 'File'
-										WHEN 'I' THEN 'Differential'
-										WHEN 'G' THEN 'Differential File'
-										WHEN 'P' THEN 'Partial Full'
-										WHEN 'Q' THEN 'Partial Differential'
-										ELSE NULL
-									  END AS Type,
-									  backupset.media_set_id AS MediaSetId,
-									  CASE mediafamily.device_type
-										WHEN 2 THEN 'Disk'
-										WHEN 102 THEN 'Permanent Disk  Device'
-										WHEN 5 THEN 'Tape'
-										WHEN 105 THEN 'Permanent Tape Device'
-										WHEN 6 THEN 'Pipe'
-										WHEN 106 THEN 'Permanent Pipe Device'
-										WHEN 7 THEN 'Virtual Device'
-										ELSE 'Unknown'
-									  END AS DeviceType,
-									  mediaset.software_name AS Software"
+						# needs compressedbackup_size for systems that support it
+						$select = "SELECT 
+					   '$servername' AS [Server],
+					    backupset.Database_Name As [Database],
+						backupset.User_Name AS Username,
+					    backupset.Backup_Start_Date as [Start],
+					    backupset.Backup_Finish_Date as [End],
+						CAST(DATEDIFF(second, backupset.backup_start_date, backupset.backup_finish_date) AS VARCHAR(4)) + ' ' + 'Seconds' as Duration,
+					    mediafamily.Physical_Device_Name AS Path,
+					    CAST((backupset.Backup_Size/1048576) AS NUMERIC(10,2)) AS TotalSizeMB,
+						CASE backupset.Type      
+				            WHEN 'L' THEN 'Log'
+				            WHEN 'D' THEN 'Full'
+				            WHEN 'F' THEN 'File'
+				            WHEN 'I' THEN 'Differential'
+				            WHEN 'G' THEN 'Differential File'
+				            WHEN 'P' THEN 'Partial Full'
+				            WHEN 'Q' THEN 'Partial Differential'
+				        ELSE NULL END AS Type, backupset.Media_Set_ID as MediaSetId, 
+					CASE mediafamily.device_type
+				                        WHEN 2 THEN 'Disk'
+				                        WHEN 102 THEN 'Permanent Disk  Device'
+				                        WHEN 5 THEN 'Tape'
+				                        WHEN 105 THEN 'Permanent Tape Device'
+				                        WHEN 6 THEN 'Pipe'
+				                        WHEN 106 THEN 'Permanent Pipe Device'
+				                        WHEN 7 THEN 'Virtual Device'
+				                        ELSE 'Unknown'
+				                        END AS DeviceType,
+					mediaset.Software_Name AS Software"
 					}
 					
-					$from = " FROM msdb..backupmediafamily mediafamily
-								 INNER JOIN msdb..backupmediaset mediaset ON mediafamily.media_set_id = mediaset.media_set_id
-								 INNER JOIN msdb..backupset backupset ON backupset.media_set_id = mediaset.media_set_id"
+					$from = " FROM msdb..BackupMediaFamily mediafamily INNER JOIN msdb..BackupMediaSet mediaset

```

Comment:
Quick question -- I remember @niphlod mentioning in this file or another file that the case of the columns mattered in CS environments (which makes sense, but damn, i always though it was just the data)

Did this get rewritten by a program that is aware of the cases of the SQL Columns in SQL Server?